src/server/pps/server/api_server.go

```diff
@@ -474,13 +473,14 @@ func (a *apiServer) UpdateJobState(ctx context.Context, request *pps.UpdateJobSt
 func (a *apiServer) UpdateJobStateInTransaction(txnCtx *txncontext.TransactionContext, request *pps.UpdateJobStateRequest) error {
 	jobs := a.jobs.ReadWrite(txnCtx.SqlTx)
 	jobPtr := &pps.StoredJobInfo{}
-	if err := jobs.Get(request.Job.ID, jobPtr); err != nil {
+	if err := jobs.Get(ppsdb.JobKey(request.Job), jobPtr); err != nil {
 		return err
 	}
 	if ppsutil.IsTerminal(jobPtr.State) {
 		return ppsServer.ErrJobFinished{Job: jobPtr.Job}
 	}
 
+	jobPtr.Started = request.Started
```

Comment:
Related to what I mentioned above, this should probably not be updateable.

Reply:
Actually, maybe it should in the restart case?

Reply:
In general, we shouldn't have code that is changing `Started` from a value to another value, this RPC just allows you to sync the state in the worker with the state in the collection transactionally with other calls.

Reply:
If we don't want to change the started field after it is first set, then we should probably remove this and lean on `ppsutil.UpdateJobState` to set it when the job is transitioned to the starting state.